bindata/scripts/load-kmod.sh

```diff
@@ -1,11 +1,16 @@
 #!/bin/sh
 # chroot /host/ modprobe $1
 kmod_name=$(tr "-" "_" <<< $1)
+kmod_args="${@:2}"
 chroot /host/ lsmod | grep "^$1" >& /dev/null
 
 if [ $? -eq 0 ]
 then
+        # NOTE: We do not check if the module is loaded with specific options
```

Comment:
unfortunately i dont have a setup to check this, 

but for a given kmod i think its parameters are under:
/sys/module/<kmod>/parameters/<param name>

what are you getting in this path for vfio ?
and what is the value you read from "enable_unsafe_noiommu_mode" file ?

from: https://man7.org/linux/man-pages/man5/sysfs.5.html

              parameters
                     This directory contains one file for each module
                     parameter, with each file containing the value of
                     the corresponding parameter.  Some of these files
                     are writable, allowing the

Reply:
moreover this can be used to validate that the provided parameters are supported.
(in addition to checking if module need to be reloaded)

Reply:
```
[root@test ~]# cat /sys/module/vfio/parameters/enable_unsafe_noiommu_mode 
Y
[root@test ~]# ls -al /sys/module/vfio/parameters/enable_unsafe_noiommu_mode 
-rw-r--r--. 1 root root 4096 Mar  8 14:50 /sys/module/vfio/parameters/enable_unsafe_noiommu_mode
```

Like I said in previous comments, I don't think we want to add complex logic to check whether or not a module is enabled with the right parameters, and should leave it to the user somehow.
Yes we could parse kernel arguments and check if the files are there but I'm sure we'll hit corner cases and cause more harm that we we think.
If you're ok, I'll not touch that code for now.

Reply:
For reference, we had this conversation here: https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/257/files#r819412615

Reply:
if there is no strong use-case for it now, im ok with deferring it.